src/rule_config.jl

```diff
@@ -14,3 +14,38 @@ function ChainRulesCore.rrule_via_ad(config::ADviaRuleConfig, f, args...; kws...
     ret === nothing && throw(MethodError(rrule, (f, args...)))
     return ret
 end
+
+# For testing this config uses finite differences to evaluate the frule and rrule
+Base.@kwdef struct ADviaFDConfig <: RuleConfig{Union{HasReverseMode, HasForwardsMode}}
+    fdm = FiniteDifferences.central_fdm(5, 1)
+end
+
+function ChainRulesCore.frule_via_ad(config::ADviaFDConfig, ȧrgs, f, args...; kws...)
+
+    call_on_copy(f, xs...) = deepcopy(f)(deepcopy(xs)...; deepcopy(kws)...)
+
+    primals_and_tangents = auto_primal_and_tangent.((f, args...))
+    primals = primal.(primals_and_tangents)
```

Comment:
Why is this not
```suggestion
    primals = (f, args...)
```
?